infra/vsts_agent_linux_startup.sh

```diff
@@ -95,6 +96,41 @@ useradd \
 #add docker group to user
 usermod -aG docker vsts
 
+# The Bazel cache fills up the hard drive in less than a day, and is a real
+# pain to clean up, as it consists of a very large amount of read-only files in
+# a fairly large tree of read-only directories. Therefore, it looks like the
+# best way to manage it is to mount it on a partition and reset the partition.
+
+CLEANUP=/home/vsts/cache_cleanup.sh
+cat <<'CACHE_CLEANUP' > $CLEANUP
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+local_cache=$HOME/.cache/bazel
+disk_cache=$HOME/.bazel-cache
+
+for file in local_cache disk_cache; do
+    eval path='$'$file
+    echo "$(date -Is) Cleaning up '$path'..."
+    if [ -d "$path/lost+found" ]; then
```

Comment:
Why do we have to check for `lost+found`? afaict, existence of `$path` should be sufficient?

Reply:
`$path` always exists; what this is trying to know is whether that's a folder on the `/` filesystem or a mount point. `ext2` filesystems have a `lost+found` directory at their root so this does the trick, though I'm happy to change that to something else if there is a better way to ask "is this folder a mount point?".

Reply:
why does `$path` always exist? I thought this check was for when we boot up and I would have expected that Bazel hasn’t created it’s cache directories at that point.

Reply:
Because we create it, right there... on the next.. ehm. Right. I see what you mean.

It used to be created before the script ran, in some previous version of the code that might have existed only in my head.